test/functional/p2p_handshake.py

```diff
@@ -60,6 +62,13 @@ def run_test(self):
         self.test_desirable_service_flags(node, [NODE_NONE, NODE_NETWORK, NODE_WITNESS], expect_disconnect=True)
         self.test_desirable_service_flags(node, [NODE_NETWORK | NODE_WITNESS], expect_disconnect=False)
 
+        self.log.info("Check that limited peers are only desired if the local chain is close to the tip (<24h)")
+        node.setmocktime(int(time.time()) + 25 * 3600)  # tip outside the 24h window, should fail
+        self.test_desirable_service_flags(node, [NODE_NETWORK_LIMITED | NODE_WITNESS], expect_disconnect=True)
+        self.generate(node, 1)  # catch up by mining a block
+        node.setmocktime(int(time.time()) + 23 * 3600)  # tip inside the 24h window, should succeed
```

Comment:
nit: This is a bit redundant. Either mining the block or changing the mock time alone will make the following test succeed.

Reply:
I think it's a bug, since this causes `ApproximateBestBlockDepth()` to return a negative number, we are not really testing the behavior of `if (ApproximateBestBlockDepth() < NODE_NETWORK_LIMITED_ALLOW_CONN_BLOCKS)`

This fixes it:
```suggestion
        self.generate(node, 1)  # catch up by mining a block
        node.bumpmocktime(23 * 3600)  # tip inside the 24h window, should succeed
```